.travis.yml

```diff
@@ -1,33 +1,66 @@
 language: cpp
 sudo: false
-compiler: clang
-#    - clang
-#    - gcc
+compiler:
+    - clang
+    - gcc
 
-os: osx
-#    - linux
-#    - osx
+os:
+    - linux
+    - osx
+#osx_image: xcode7
+dist: trusty
+
+addons:
+  apt: 
+   sources:
+      - boost-latest
+      - ubuntu-toolchain-r-test
+   packages:
+      - gcc-4.8
+      - g++-4.8
+      - libboost1.55-all-dev
+      - libilmbase-dev
+      - libopenexr-dev
+      - libgif-dev
+      - libtiff4-dev
+      - libopenjpeg-dev
+      - libwebp-dev
+      - ffmpeg
 
 cache:
     ccache: true
+    apt: true
     directories:
     - $HOME/.ccache
     - $HOME/lgritz/oiio-images
     - $HOME/lgritz/libtiffpic
 
 before_install:
-    - src/travis/before_install.bash
+    - if [ $TRAVIS_OS_NAME == osx ] ; then sysctl machdep.cpu.features ; fi
+    - if [ $TRAVIS_OS_NAME == linux ] ; then cat /proc/cpuinfo ; fi
 
 install:
-    - src/travis/install.bash
+    - if [ "$CXX" == "g++" ]; then export CXX="g++-4.8" ; fi
+    - if [ $TRAVIS_OS_NAME == osx ] ; then src/build-scripts/install_homebrew_deps.bash ; fi
+    - if [ $TRAVIS_OS_NAME == linux ] ; then src/build-scripts/travis_build_openexr.bash ; fi
+    - if [ $TRAVIS_OS_NAME == linux ] ; then export ILMBASE_HOME=${HOME}/openexr-install ; export OPENEXR_HOME=${HOME}/openexr-install ; fi
+    - src/build-scripts/install_test_images.bash
 
 # before_script:
 
-script: make VERBOSE=1 cmakesetup && make -j 2 && make test
+# Strategy: make a full default build + test (default to sse2 and C++03)
+# Then just check that the build works fine for sse4.2.
+# And finally, do a build and test with C++11.
+script:
+    - make VERBOSE=1 cmakesetup && make -j 2 && make test
+    - make nuke && make -j 2 USE_SIMD=sse4.2 && make test
+    - make nuke && make -j 2 USE_CPP11=1 && make test
```

Comment:
Pardon the ignorance, but what about using [ENV variables](https://docs.travis-ci.com/user/speeding-up-the-build/) for the flags? 

For example, 
```
env:
  - BUILD_FLAGS=""
  - BUILD_FLAGS="USE_SIMD=sse4.2"
  - BUILD_FLAGS="USE_CPP11=1"

script:
  - make VERBOSE=1 cmakesetup
  - make -j2 $BUILD_FLAGS 
  - make test
```

This would both grow the build matrix by an order of 3, and it would also run the containers in parallel. It would need to be tested by the joys of trial and error, but should work.